Source/Initialization/InitSpaceChargeField.cpp

```diff
@@ -120,25 +149,136 @@ WarpX::computeE(amrex::Vector<std::array<std::unique_ptr<amrex::MultiFab>, 3> >&
             const auto& Ey_arr = (*E[lev][1])[mfi].array();
             const auto& Ez_arr = (*E[lev][2])[mfi].array();
 
+            Real beta_x = beta[0];
+            Real beta_y = beta[1];
+            Real beta_z = beta[2];
+
+            // Calculate the electric field
+            // Use discretized derivative that matches the staggering of the grid.
 #if (AMREX_SPACEDIM == 3)
             amrex::ParallelFor( tbx, tby, tbz,
                 [=] AMREX_GPU_DEVICE (int i, int j, int k) {
-                    Ex_arr(i,j,k) += -inv_dx*( phi_arr(i+1,j,k) - phi_arr(i,j,k) );
+                    Ex_arr(i,j,k) +=
+                        +(beta_x*beta_x-1)*inv_dx*( phi_arr(i+1,j,k)-phi_arr(i,j,k) )
+                        +beta_x*beta_y*0.5*inv_dy*( phi_arr(i,j+1,k)-phi_arr(i,j-1,k) )
+                        +beta_x*beta_z*0.5*inv_dz*( phi_arr(i,j,k+1)-phi_arr(i,j,k-1) );
```

Comment:
For the dy and terms, I think, phi at i+1 need to be used too.

Reply:
Ah, yes, that's right. I'll update the PR accordingly.